test/checks/shared/duplicate-id.js

```diff
@@ -58,4 +58,52 @@ describe('duplicate-id', function () {
 		assert.isTrue(checks['duplicate-id'].evaluate.call(checkContext, node));
 	});
 
+	(shadowSupport.v1 ? it : xit)('should find duplicate IDs in shadow trees', function () {
+		var div = document.createElement('div');
+		div.id = 'target';
+		var shadow = div.attachShadow({ mode: 'open' });
+		shadow.innerHTML = '<span id="target"></span><p id="target">text</p>';
+		var node = shadow.querySelector('span');
+		fixture.appendChild(div);
+
+		assert.isFalse(checks['duplicate-id'].evaluate.call(checkContext, node));
+		assert.lengthOf(checkContext._relatedNodes, 1);
+		assert.deepEqual(checkContext._relatedNodes, [shadow.querySelector('p')]);
+	});
+
+	(shadowSupport.v1 ? it : xit)('should ignore same IDs in shadow trees', function () {
+		var node = document.createElement('div');
+		node.id = 'target';
+		var shadow = node.attachShadow({ mode: 'open' });
+		shadow.innerHTML = '<span id="target"></span>';
+		fixture.appendChild(node);
+
+		assert.isTrue(checks['duplicate-id'].evaluate.call(checkContext, node));
+		assert.lengthOf(checkContext._relatedNodes, 0);
+	});
+
+	(shadowSupport.v1 ? it : xit)('should ignore same IDs outside shadow trees', function () {
+		var div = document.createElement('div');
+		div.id = 'target';
+		var shadow = div.attachShadow({ mode: 'open' });
+		shadow.innerHTML = '<span id="target"></span>';
+		var node = shadow.querySelector('#target');
+		fixture.appendChild(div);
+
+		assert.isTrue(checks['duplicate-id'].evaluate.call(checkContext, node));
+		assert.lengthOf(checkContext._relatedNodes, 0);
+	});
+
+	(shadowSupport.v1 ? it : xit)('should not ignore slotted elements', function () {
+		var node = document.createElement('div');
+		node.id = 'target';
+		node.innerHTML = '<p id="target">text</p>';
+		var shadow = node.attachShadow({ mode: 'open' });
+		shadow.innerHTML = '<span id="target"><slot></slot></span>';
+		fixture.appendChild(node);
+
+		assert.isFalse(checks['duplicate-id'].evaluate.call(checkContext, node));
+		assert.lengthOf(checkContext._relatedNodes, 1);
+		assert.deepEqual(checkContext._relatedNodes, [node.querySelector('p')]);
```

Comment:
I like the description of the test because it describes what is actually happening i.e. that the slotted content (light DOM) is being compared with the light DOM and not the shadow DOM it is slotted into. You could slightly clarify it by saying

```
Compare slotted content with the light DOM
```